frontend/pages/product_new_page.js

```diff
@@ -77,6 +81,7 @@ export default class AddProductForm extends PureComponent {
 
   addProduct = (e) => {
     // e.preventDefault();
+    if(this.state.dataInputStatus == '') this.state.dataInputStatus = 'incomplete';
```

Comment:
couple of things here:
1. space after if `if (`
2. you can just check if it's falsey `if (!this.state.dataInputStatus) ...`
3. if you did need to do an equality check, you'd need triple equal signs
4. you shouldn't assign to state directly, remember that updating state is an async process and you have to use `this.setState`
5. in addition to setState being async, it also batches state changes together from different parts of the component for performance so that state is not updated multiple times, and render is not called multiple times

Reply:
fixed